docs/RFCS/sql_batching.md

```diff
@@ -0,0 +1,499 @@
+- Feature Name: SQL Statement Batching
+- Status: draft
+- Start Date: 2017-01-18
+- Authors: Nathan VanBenschoten
+- RFC PR: [#13160](https://github.com/cockroachdb/cockroach/issues/13160)
+- Cockroach Issue: [#10057](https://github.com/cockroachdb/cockroach/issues/10057),
+                   [#2210](https://github.com/cockroachdb/cockroach/issues/2210),
+                   [#4403](https://github.com/cockroachdb/cockroach/issues/4403),
+                   [#5259](https://github.com/cockroachdb/cockroach/issues/5259)
+
+# Summary
+
+This RFC proposes a method to support the batching of multiple SQL statements for parallel
+execution within a single transaction. If two or more adjacent SQL statements are provably
+independent, batching their KV operations and running them together can yield up to a linear
+speedup. This will serve as a critical optimization in cases where replicas have
+high link latencies and round-trip communication costs dominates system performance.
+
+
+# Motivation
+
+In positioning CockroachDB as a globally-distributed database, high communication
+latency between nodes in Cockroach clusters is likely to be a reality for many
+real-world deployments. This use case is, therefore, one that we should make sure
+to handle well. In this RFC, we focus on an individual aspect of this larger goal:
+reducing aggregate latency resulting from high communication cost for SQL
+transactions with multiple independent statements.
+
+A SQL transaction is generally composed of a series of SQL statements. In some
+cases, these statements depend on one another and rely on strictly ordered serial
+execution for correctness. However, in many other cases, the statements are either
+partially or fully independent of one another. This independence between statements
+provides an opportunity to exploit [intrinsic intra-statement parallelism](#independent-statements),
```

Comment:
Do you mean "inter-statement" parallelism? "Intra-statement" means "within a single statement"